### DISCUSSION TITLE
Quarto on Red Hat linux

### DISCUSSION THREAD

**Discussion by tomkom** [quarto-dev/quarto-cli#380] [Q&A]
We are trying to run Quarto on a system used by a number of students, running Jupyterlab and Python on Anaconda.
The system is on RH - a constraint. We have tried both a) convert the deb file to rpm and b) compile from the source code. Both approaches gets the same error that the package needs to have GLIBC v2.18. The problem is the RH7 provides GLIBC v2.17 and the RH8 provides v2.28.  We haven’t found a way to compile the package with v2.18 yet.

Would anyone be able to share experiences with Quarto on RH?

**Comment by jjallaire** [comment]
The core issue here is that Deno (our JavaScript runtime) doesn't work on RHEL 7. Here's the relevant issue in the Deno repo: https://github.com/denoland/deno/issues/1658

**Comment by sbamin** [comment]
Until we find a permanent fix for RHEL, Cent OS 7 system, I am able to run `quarto` without GLIB error on our HPC running Cent OS 7.

I replace default `deno` file at `<path to quarto .tar.gz extracted code>/share/deno` with a wrapper that uses deno container image. See https://gist.github.com/sbamin/be2505732702d8b729ed1a0d0216e700

I have tested installation as follows:

```sh
module load quarto

# for now, quarto fails to check python without giving QUARTO_PYTHON
# variable
export QUARTO_PYTHON=<path to python3 in one of conda env with jupyter and R installed>

quarto check
```

```
[✓] Checking Quarto installation......OK
      Version: 0.9.254
      Path: <USER_HPCENV>/opt/modules/apps/quarto/0.9/bin

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.9.9 (Conda)
      Path: <USER_HPCENV>/mambaforge/envs/rey/bin/python
      Jupyter: 4.9.1
      Kernels: rey_r41, ben_py38, yoda_py310, rey_py39, yoda_bash, yoda_r41, yoda_jl16, ir, python3

(-) Checking Jupyter engine render..../bin/bash: <USER_HPCENV>/mambaforge/envs/rey/lib/libtinfo.so.6: no version information available (required by /bin/bash)
<USER_HPCENV>/opt/kernels/wrap_ben_py38: line 17: conda: command not found
[✓] Checking Jupyter engine render....OK

(|) Checking R installation...........During startup - Warning messages:
1: Setting LC_CTYPE failed, using "C" 
2: Setting LC_COLLATE failed, using "C" 
3: Setting LC_TIME failed, using "C" 
4: Setting LC_MESSAGES failed, using "C" 
5: Setting LC_MONETARY failed, using "C" 
6: Setting LC_PAPER failed, using "C" 
7: Setting LC_MEASUREMENT failed, using "C" 
[✓] Checking R installation...........OK
      Version: 4.1.1
      Path: <USER_HPCENV>/mambaforge/envs/rey/lib64/R
      LibPaths:
        - <USER_HPCENV>/opt/R/pkgs/rey4.1
        - <USER_HPCENV>/mambaforge/envs/rey/lib/R/library
        - <USER_HPCENV>/opt/R/pkgs/4.1
        - <USER_HPCENV>/mambaforge/envs/yoda/lib/R/library
      rmarkdown: 2.11

[✓] Checking Knitr engine render......OK
```

I am yet to test if I can render python and R code on HPC using this work around.
